src/Evolution/Systems/ForceFree/RecoverPhysicalVariables.cpp

```diff
@@ -12,22 +12,31 @@
 
 namespace ForceFree::Tags {
 
+namespace {
+void divide_vector_with_sqrt_gamma(
+    const gsl::not_null<tnsr::I<DataVector, 3>*> vector,
+    const tnsr::I<DataVector, 3>& densitized_vector,
+    const Scalar<DataVector>& sqrt_det_spatial_metric) {
+  get<0>(*vector) = get<0>(densitized_vector) / get(sqrt_det_spatial_metric);
+  get<1>(*vector) = get<1>(densitized_vector) / get(sqrt_det_spatial_metric);
+  get<2>(*vector) = get<2>(densitized_vector) / get(sqrt_det_spatial_metric);
+}
+}  // namespace
+
 void ElectricFieldCompute::function(
     const gsl::not_null<tnsr::I<DataVector, 3>*> electric_field,
-    const typename tnsr::I<DataVector, 3>& tilde_e,
+    const tnsr::I<DataVector, 3>& tilde_e,
     const Scalar<DataVector>& sqrt_det_spatial_metric) {
-  for (size_t d = 0; d < 3; ++d) {
-    (*electric_field).get(d) = tilde_e.get(d) / get(sqrt_det_spatial_metric);
-  }
+  divide_vector_with_sqrt_gamma(electric_field, tilde_e,
```

Comment:
What I meant was something like this:

```cpp
void electric_field(...) {
  tenex::evaluate<ti::I>(result, tilde_e(ti::I) / sqrt_gamma()); // or your `divide_vector` function
}

// In .hpp
struct ElectricFieldCompute : ElectricField, db::ComputeTag {
  // ...
  static constexpr auto function = &electric_field;
};
```

This way you have the functions implemented independently of the compute tags, and the compute tags just use them.